test/test_car.py

```diff
@@ -0,0 +1,91 @@
+import pytest
+import time
+import numpy as np
+
+from vehicle_simulation_environment import car
+from scipy.integrate import quad
+
+
+TOLERANCE = 0.05
+
+
+class TestCar(object):
+    @pytest.mark.parametrize(
+        "x,y,theta,vel,accel,ang_vel,ang_accel,delta_accel,delta_ang_accel",
+        [
+            (0, 0, 0, 0, 0, 0, 0, 0, 0),
+            (0, 0, 1, 2, 0, 0, 0, 0, 0),
+            (1, -1, 0.5, 0.75, 0, 0, 0, 0, 0),
+            (0, 0, 1, 0, 0, 3, -2.5, 0, 0),
+            (-1, 1, 2.1, -1, 3, 1, -1, 0, 0),
+            (1, 1, 0, 0, 0, 2, -1, 1.5, 0),
+            (2, 4, 5, 2, 1, 1, 5, 0, -2),
+            (5, 2, 4, -3, -2, -5, 2, 2, 3),
+            (-2.56, 0.1, 0.2, 2.1, -0.02, 0.6, -1, 3, 2),
+            (200, 200, -15, 10, 10, -10, -10, 5, -5),
+        ],
+    )
+    def test_movement(
+        self, x, y, theta, vel, accel, ang_vel, ang_accel, delta_accel, delta_ang_accel
+    ):
+
+        def cal_expected_val(params):
```

Comment:
There's a lot of magic numbers flying around here, let's use a dataclass instead.